External DNS is failing in Oracle Cloud OKE with version v1.33.1

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

Make sure to validate the behavior against latest release https://github.com/kubernetes-sigs/external-dns/releases as we don't support past versions.
-->

**What happened**:
External DNS pod is failing is continuously failing (CrashLoopBackOff) with following error message. I tried with version v0.18.0 and v0.19.0. Same error message.

time="2025-09-18T18:00:02Z" level=info msg="GitCommitShort=unknown, GoVersion=go1.24.4, Platform=linux/amd64, UserAgent=ExternalDNS/v20250626-v0.18.0"
time="2025-09-18T18:00:02Z" level=info msg="Instantiating new Kubernetes client"
time="2025-09-18T18:00:02Z" level=info msg="Using inCluster-config based on serviceaccount-token"
time="2025-09-18T18:00:02Z" level=info msg="Created Kubernetes client https://10.96.0.1:443"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x315a9d1]

goroutine 1 [running]:
sigs.k8s.io/external-dns/controller.buildProvider({0x44b7480, 0xc000358140}, 0xc00019d208, 0xc00062a380)
        sigs.k8s.io/external-dns/controller/execute.go:286 +0x471
sigs.k8s.io/external-dns/controller.Execute()
        sigs.k8s.io/external-dns/controller/execute.go:110 +0x3ee
main.main()
        sigs.k8s.io/external-dns/main.go:26 +0xf


**What you expected to happen**:
External DNS deployment should be running fine.

**How to reproduce it (as minimally and precisely as possible)**:

Used the steps from link
https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/oracle.md

<!--
Please provide as much detail as possible, including Kubernetes manifests with spec.status, ExternalDNS arguments, and logs. A bug that cannot be reproduced won't be fixed.
-->

**Anything else we need to know?**:

**Environment**:

- External-DNS version (use `external-dns --version`): 0.18.0 and 0.19.0
- DNS provider: Oracle Cloud DNS
- Others:

--- a/controller/execute.go
+++ b/controller/execute.go
@@ -278,17 +278,17 @@ func buildProvider(
 		if cfg.OCIAuthInstancePrincipal {
 			if len(cfg.OCICompartmentOCID) == 0 {
 				err = fmt.Errorf("instance principal authentication requested, but no compartment OCID provided")
-			} else {
-				authConfig := oci.OCIAuthConfig{UseInstancePrincipal: true}
-				config = &oci.OCIConfig{Auth: authConfig, CompartmentID: cfg.OCICompartmentOCID}
+				break
 			}
+			authConfig := oci.OCIAuthConfig{UseInstancePrincipal: true}
+			config = &oci.OCIConfig{Auth: authConfig, CompartmentID: cfg.OCICompartmentOCID}
 		} else {
-			config, err = oci.LoadOCIConfig(cfg.OCIConfigFile)
+			if config, err = oci.LoadOCIConfig(cfg.OCIConfigFile); err != nil {
+				break
+			}
 		}
 		config.ZoneCacheDuration = cfg.OCIZoneCacheDuration
-		if err == nil {
-			p, err = oci.NewOCIProvider(*config, domainFilter, zoneIDFilter, cfg.OCIZoneScope, cfg.DryRun)
-		}
+		p, err = oci.NewOCIProvider(*config, domainFilter, zoneIDFilter, cfg.OCIZoneScope, cfg.DryRun)
 	case "rfc2136":
 		tlsConfig := rfc2136.TLSConfig{
 			UseTLS:                cfg.RFC2136UseTLS,

--- a/controller/execute_test.go
+++ b/controller/execute_test.go
@@ -265,6 +265,23 @@ func TestBuildProvider(t *testing.T) {
 			expectedType: "*provider.CachedProvider",
 		},
 		{
+			name: "oci provider instance principal without compartment OCID",
+			cfg: &externaldns.Config{
+				Provider:                 "oci",
+				OCIAuthInstancePrincipal: true,
+				OCICompartmentOCID:       "",
+			},
+			expectedError: "instance principal authentication requested, but no compartment OCID provided",
+		},
+		{
+			name: "oci provider without config file",
+			cfg: &externaldns.Config{
+				Provider:      "oci",
+				OCIConfigFile: "",
+			},
+			expectedError: "reading OCI config file",
+		},
+		{
 			name: "coredns provider",
 			cfg: &externaldns.Config{
 				Provider: "coredns",

